contracts/migration/SushiMigrator.sol

```diff
@@ -0,0 +1,189 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.10;
+
+import "../libraries/SafeERC20.sol";
+import "../interfaces/ITreasury.sol";
+import "../interfaces/IUniswapV2Pair.sol";
+import "../interfaces/IUniswapV2Router.sol";
+import "../types/OlympusAccessControlledV2.sol";
+
+/// @notice This contract handles migrating lps from sushiswap to uniswap v2
+///         to make use of this contract ensure contract address is RESERVEMANAGER IN OHM Treasury
+contract SushiMigrator is OlympusAccessControlledV2 {
+    using SafeERC20 for IERC20;
+
+    struct TxDetails {
+        uint256 balanceALeft;
+        uint256 balanceBLeft;
+        uint256 contractV2lpBalanceBeforeRemovingLiquidity;
+        uint256 contractV2lpBalanceAfterRemovingLiquidity;
+        uint256 expectedToken0ToBeAddedOnUniswapV2;
+        uint256 expectedToken1ToBeAddedOnUniswapV2;
+        uint256 lp;
+    }
+
+    uint256 public txCount;
+
+    mapping(uint256 => TxDetails) public transactions;
+
+    address immutable treasury = 0x9A315BdF513367C0377FB36545857d12e85813Ef;
+
+    constructor(address _olympusAuthority) OlympusAccessControlledV2(IOlympusAuthority(_olympusAuthority)) {}
+
+    /// @notice Takes lp from treasury, remove liquidity from sushi, adds liquidity to uniswap v2
+    /// @param sushiRouter_ sushi router addres
+    /// @param uniRouter_ uni router address
+    /// @param sushiLpAddress_ sushi lp address
+    /// @param uniswapLpAddress_ uni lp address
+    /// @param amount_ lp amount to get from treasury and amount of liquidity to remove
+    /// @param removeLiquiditySlippage_ slippage used when removing liquidity....i.e 95% will be 950
+    /// @param addLiquiditySlippage_ slippage used when adding liquidity....i.e 95% will be 950
+    function executeTx(
+        address sushiRouter_,
+        address uniRouter_,
+        address sushiLpAddress_,
+        address uniswapLpAddress_,
+        uint256 amount_,
+        uint256 removeLiquiditySlippage_,
+        uint256 addLiquiditySlippage_
+    ) external onlyGovernor {
+        ITreasury(treasury).manage(sushiLpAddress_, amount_);
+
+        uint256 amount = IUniswapV2Pair(sushiLpAddress_).balanceOf(address(this));
+
+        (uint256 amountOHM, uint256 amountToken) = removeLiquidity(
```

Comment:
Is token0 always OHM? I know that this is not relevant to how the transaction develops further but it induces confusion too.

Reply:
Yes in this case, given this is focus on OHM/DAI and OHM/ETH pools.